x-pack/platform/plugins/private/intercepts/server/services/intercept_trigger.ts

```diff
@@ -84,62 +85,72 @@ export class InterceptTriggerService {
       isRecurrent?: boolean;
     }
   ) {
-    const existingTriggerDefinition = await this.fetchRegisteredTask(triggerId);
-
-    const { triggerAfter, isRecurrent } = cb({
-      existingTriggerDefinition,
-    });
+    try {
+      const existingTriggerDefinition = await this.fetchRegisteredTask(triggerId);
 
-    if (!triggerAfter) {
-      this.logger?.error('Trigger interval is not defined');
-      return;
-    }
+      const { triggerAfter, isRecurrent } = cb({
+        existingTriggerDefinition,
+      });
 
-    if (!existingTriggerDefinition) {
-      const contextName = `productInterceptTriggerCreation:${triggerId}` as const;
+      if (!triggerAfter) {
+        this.logger?.error('Trigger interval is not defined');
+        return;
+      }
 
-      await this.savedObjectsClient
-        ?.create<InterceptTriggerRecord>(
-          this.savedObjectRef.name,
-          {
-            firstRegisteredAt: new Date().toISOString(),
+      if (!existingTriggerDefinition) {
+        const contextName = `productInterceptTriggerCreation:${triggerId}` as const;
+
+        await this.savedObjectsClient
+          ?.create<InterceptTriggerRecord>(
+            this.savedObjectRef.name,
+            {
+              firstRegisteredAt: new Date().toISOString(),
+              triggerAfter,
+              installedOn: this.kibanaVersion!,
+              recurrent: isRecurrent ?? true,
+            },
+            { id: triggerId }
+          )
+          .then((result) => {
+            this.counter?.usageCounter(contextName);
+            return result;
+          })
+          .catch((err) => {
+            if (SavedObjectsErrorHelpers.isConflictError(err)) {
+              this.logger?.debug(`Conflict error creating registered task: ${err.message}`);
+              return null;
+            }
+
+            this.counter?.errorCounter(contextName);
+            throw err;
+          });
+        return;
+      } else if (
+        // only support updating the trigger interval for existing trigger definitions
+        existingTriggerDefinition &&
+        existingTriggerDefinition.triggerAfter !== triggerAfter
```

Comment:
I suggest we remove this - it seems that trigger state becomes broken if you update the interval since math doesn't add up anymore. please double check, maybe I am missing something